std/typecons.d

```diff
@@ -3431,6 +3431,36 @@ Params:
     assert(!npi.isNull);
 }
 
+	/**
+      If they are both null, then they are equal. If one is null and the other
+      is not, then they are not equal. If they are both non-null, then they are
+      equal if their values are equal.
+      */
+    bool opEquals()(auto ref const(typeof(this)) rhs) const
+    {
+        if (isNull)
+            return rhs._isNull;
+        if (rhs._isNull)
+            return false;
+        return _value == rhs._value;
+    }
+
+    /// Ditto
+    bool opEquals(U)(auto ref const(U) rhs) const
+    if (is(typeof(this.get == rhs)))
```

Comment:
This doesn't seem right. Shouldn't it be:
````
if (is(typeof(this.get) == typeof(rhs)))
````
?

Reply:
That is possible, it's taken from `Nullable(T)` which means the issue will exist within both types.

Will change it for both if that would be correct. Will wait for a more official answer to that.

Reply:
No, it's right, it's asking can `this.get() == rhs` compile.

You wouldn't want the other way, because then e.g. `Nullable!int` couldn't be compared to `1L`

Reply:
I think this is correct, but it may be surprising in some cases. For example:
```D
enum NullValue = int.max;
Nullable!(int, NullValue) n;

assert(n == NullValue); // fails
```

However, I also don't know that we want to make this true either. I think the proposed behavior is at least consistent with the other Nullable.

Reply:
At least should be `__traits(compiles, this.get == rhs)` to avoid ambiguity, right?

Reply:
Can you explain the ambiguity? I'm always lost when it comes to the differences between `__traits(compiles` and `is(typeof`

Reply:
I mean, to avoid the confusion with `is(typeof() == typeof())`.

Reply:
But there *are* subtle differences. I don't remember what they are, but someone (maybe Timon or Stefan?) has outlined why you should prefer one or the other, I don't have the link to the NG post handy. I'd prefer leaving it the way it is for this PR, since it's already that way for the other version of `Nullable`, and we can explore changing the `is(typeof` to `__traits(compiles` across the whole module in another PR.